### PR TITLE
feat: Activestorage direct upload

### LINKED ISSUE #3567: feat: Ability to use active storage direct upload
Large file uploads could clog up the application server. They are also susceptible to requesting time outs from services like Heroku. So let's explore implementing file uploads via active storage direct upload

ref: 
https://edgeguides.rubyonrails.org/active_storage_overview.html#direct-uploads
https://blog.commutatus.com/understanding-rails-activestorage-direct-uploads-a4aeca7eccf
https://devcenter.heroku.com/articles/request-timeout
https://www.bigbinary.com/blog/uploading-files-directly-to-s3-using-pre-signed-post-request
https://devcenter.heroku.com/articles/direct-to-s3-image-uploads-in-rails

### PR DESCRIPTION
**Pull request by tejaswinichile**
# Pull Request Template

## Description

Fixes #3567 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Tested file image upload through the reply box from the dashboard.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by tejaswinichile** [comment]
@fayazara / @muhsin-k / @iamsivin Can you help me to make these vue changes according to the standards that we follow.

File: app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue [hunk 1]
```diff
@@ -439,6 +440,35 @@ export default {
...
+    onDirectFileUpload(file) {
+      if (!file) {
+        return;
+      }
+      if (checkFileSizeLimit(file, MAXIMUM_FILE_UPLOAD_SIZE)) {
+        const upload = new DirectUpload(file.file, '/rails/active_storage/direct_uploads', null, file.file.name);
+        upload.create((error, blob) => {
+          if (error) {
+            this.showAlert(
+              "Not able to upload the file."
```
Comment (iamsivin) [inline app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:452]: Use i18n here

File: app/javascript/widget/components/ChatAttachment.vue [hunk 2]
```diff
@@ -44,6 +47,39 @@ export default {
...
+      }
+      this.isUploading = true;
+      try {
+        if (checkFileSizeLimit(file, MAXIMUM_FILE_UPLOAD_SIZE)) {
+          const upload = new DirectUpload(file.file, '/rails/active_storage/direct_uploads', null, file.file.name);
+
+          upload.create((error, blob) => {
+            if (error) {
+              window.bus.$emit(BUS_EVENTS.SHOW_ALERT, {
+                message: this.$t("Not able to upload the file."),
```
Comment (iamsivin) [inline app/javascript/widget/components/ChatAttachment.vue:62]: Use i18n here also

File: app/javascript/dashboard/api/inbox/message.js [hunk 3]
```diff
@@ -14,10 +14,11 @@ export const buildCreatePayload = ({
   let payload;
   if (file) {
     payload = new FormData();
-    payload.append('attachments[]', file, file.name);
+    // payload.append('attachments[]', file, file.name);
```
Comment (iamsivin) [inline app/javascript/dashboard/api/inbox/message.js:17]:
```suggestion

```

**Review by nithindavid — COMMENTED** [review]
FE part looks good! Good to merge after fixing the remaining comments

File: app/javascript/dashboard/api/inbox/message.js (hunk 3 above)
Comment (muhsin-k) [inline app/javascript/dashboard/api/inbox/message.js:17]: Remove the commented code

File: app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue (hunk 1 above)
Comment (muhsin-k) [inline app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:452]: Show error message if error object contains any message or show default message

File: package.json [hunk 4]
```diff
@@ -76,11 +77,11 @@
...
     "@babel/preset-env": "7.13.15",
     "@storybook/addon-actions": "6.2.9",
-    "@storybook/addon-docs": "^6.2.9",
-    "@storybook/addon-essentials": "6.2.9",
+    "@storybook/addon-docs": "^6.4.9",
+    "@storybook/addon-essentials": "^6.4.9",
     "@storybook/addon-links": "6.2.9",
     "@storybook/addons": "6.2.9",
-    "@storybook/vue": "6.2.9",
+    "@storybook/vue": "^6.4.9",
```
[the remark this answers is not in the document]
Comment (tejaswinichile) [reply package.json:84]: No not intentional, It got updated with activestorage.js

**Comment by tejaswinichile** [comment]
@muhsin-k I have resolved the comments please have a look.

File: app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue [hunk 5]
```diff
@@ -443,6 +444,35 @@ export default {
...
     },
+    onDirectFileUpload(file) {
+      if (!file) {
+        return;
+      }
+      if (checkFileSizeLimit(file, MAXIMUM_FILE_UPLOAD_SIZE)) {
+        const upload = new DirectUpload(file.file, '/rails/active_storage/direct_uploads', null, file.file.name);
+        upload.create((error, blob) => {
+          if (error) {
+            this.showAlert(
```
Comment (muhsin-k) [inline app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:455]: @tejaswinichile Are you sure the type of error is a string?

File: app/javascript/widget/components/ChatAttachment.vue [hunk 6]
```diff
@@ -44,6 +47,39 @@ export default {
...
+      }
+      this.isUploading = true;
+      try {
+        if (checkFileSizeLimit(file, MAXIMUM_FILE_UPLOAD_SIZE)) {
+          const upload = new DirectUpload(file.file, '/rails/active_storage/direct_uploads', null, file.file.name);
+
+          upload.create((error, blob) => {
+            if (error) {
+              window.bus.$emit(BUS_EVENTS.SHOW_ALERT, {
+                message: this.$t(error),
```
Comment (muhsin-k) [inline app/javascript/widget/components/ChatAttachment.vue:62]: Seems like we have to define `error` key in i18n config

File: app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue (hunk 5 above)
Comment (tejaswinichile) [reply app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:455]: Yes, I checked. Let me add the screenshot for the same.

**Comment by tejaswinichile** [comment]
@muhsin-k Errors are showing fine at both the places, where we changed the direct upload.
img2
image1

**Review by muhsin-k — APPROVED** [review]

File: app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue [hunk 7]
```diff
@@ -61,7 +61,7 @@
     <reply-bottom-panel
       :mode="replyType"
       :send-button-text="replyButtonLabel"
-      :on-file-upload="onFileUpload"
```
Comment (tejaswinichile) [inline app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:64]: @pranavrajs Can we make this conditional?

**Pull request merged**